File: src/main/java/com/catboyranch/roborancher/commands/CCat.java
package com.catboyranch.roborancher.commands;

import com.catboyranch.roborancher.utils.RoleType;
import com.catboyranch.roborancher.utils.CommandArgument;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import com.catboyranch.roborancher.RoboRancher;
import com.catboyranch.roborancher.Server;
import com.catboyranch.roborancher.utils.Utils;

import javax.imageio.ImageIO;
import java.io.File;
import java.net.URL;

public class CCat extends CommandBase{
    private final static String IMAGE_URL = "https://thiscatdoesnotexist.com";

    public CCat(RoboRancher rancher) {
        super(rancher);
        name = "cat";
        description = "Cat";
        allowedRoles = Utils.asArray(RoleType.ADMIN, RoleType.MODERATOR, RoleType.MEMBER);
    }

    @Override
    public void run(String cmd, Server server, MessageReceivedEvent event, CommandResult result, CommandArgument... args) {
        try {
            File file = File.createTempFile("cat", ".png");
            ImageIO.write(ImageIO.read(new URL(IMAGE_URL)), "png", file);
            event.getChannel().sendFile(file).queue();
        } catch (Exception e) {
            result.error(String.format("Error fetching image. (%s)", e.getMessage()));
            return;
        }
        result.success();
    }

}

File: src/main/kotlin/com/catboyranch/roborancher/commands/CCat.kt
package com.catboyranch.roborancher.commands

import com.catboyranch.roborancher.RoboRancher
import com.catboyranch.roborancher.RoleUtils
import com.catboyranch.roborancher.Server
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import java.io.File
import java.net.URL
import javax.imageio.ImageIO

class CCat(rancher: RoboRancher): CommandBase(rancher) {
    override val name = "cat"
    override val description = "Cat"
    override val allowedRoles = arrayOf(RoleUtils.RoleType.ADMIN, RoleUtils.RoleType.MODERATOR, RoleUtils.RoleType.MEMBER)

    override fun run(cmd: String, server: Server, event: MessageReceivedEvent, result: CommandResult, vararg args: CommandArgument) {
        try {
            val file = File.createTempFile("cat", ".png")
            ImageIO.write(ImageIO.read(URL("https://thiscatdoesnotexist.com")), "png", file)
            event.channel.sendFile(file).queue()
        } catch (exception: Throwable) {
            result.error("Error fetching image! Reason: ${exception.message}")
            return
        }
        result.success()
    }
}